FRC1710-2019/src/main/java/frc/robot/Vision.java

```diff
@@ -0,0 +1,111 @@
+        
+        //just prints the "margin of error" to the console log
+        
+        double yRad = Math.toRadians(y);
+        double tanOfY = Math.tan(yRad);
+        double distance = (Constants.Hdiff/tanOfY)/10;
+        System.out.println(allowableRange(Constants.MAX, Constants.con, distance));
+
+        //if A button is pressed
+        if(Drive.driveStick.getRawButton(1) == true){
+            
+          //whatup my diggity dawg
+          //read the varible name ;)
+            double turnPower = 0.0;
+
+            /*what this block is going to do is affect the turn power values to 
+            arcade drive to allow for user input until it sees a target
+            if a target is in range and satifies the thresholds then stop allowing a
+            user input and lock on to the crosshairs of the target, 
+            fwd power is always dictated by the driver*/
+
+            if(IsActualTarget(t) == true && IsSatified(x, y) == true) { //if satisfied within range
+                turnPower = 0.0; //dont turn
+            }else if(IsActualTarget(t) == true && IsInRange(x, y) == true){//if it can see a target and its in a certain range
+                turnPower = (Constants.kpAim * x); //if a target is within a certain range turn smothly this is a p loop
+            }else if(IsActualTarget(t) == false){
+                turnPower = -Drive.getTurnPower() * .2; //if there is no target allow for user input
+            }
+
+            Drive.arcadeDrive(turnPower, Drive.getForwardPower() * .35); //apply the values to robot
+        } 
+        
+    }      
+      
+      
+      public static boolean IsActualTarget(double Is_target) { //can the limelight see a target
+        if(Is_target >= 1){return true;} 
+        else{return false;} 
+      }
+      public static boolean IsSatified(double target_x, double target_y) { //are we happy? 
+        //make limelight entries local 
+        NetworkTable table = NetworkTableInstance.getDefault().getTable("limelight");
+        NetworkTableEntry ty = table.getEntry("ty");
+        double y = ty.getDouble(0.0);
+
+        //this finds the distance between the target and the limelight
+        double yRad = Math.toRadians(y);
+        double tanOfY = Math.tan(yRad);
+        double distance = (Constants.Hdiff/tanOfY)/10; 
+
+        if(Math.abs(target_x) <= allowableRange(Constants.MAX, Constants.con, distance)){return true;} 
+        else{return false;} 
+      }
+
+      public static boolean IsInRange(double target_x, double target_y) { //are we in range of a target?
+        if(Math.abs(target_x) >= 10.0){return true;} 
+        else{return false;} 
+      }
+      
+      public static double allowableRange(double max, double c, double distance){
+        // ight kiddos, get ready
+        //this equasion gives us a "smooth" down turn to zero for our "margin of error" 
+        //based on distance
+        // this is the equasion (-max * 2^k*distance + max)
```

Comment:
Nit: Code quality is important. This comment should be above the function def. so it can be pulled in by the docs/autocomplete by VSCode when used.

e.g. 
```
/**
 * 'Ight kiddos, get ready. 
 * This equation gives us a smooth turn (down for what) for our "margin of error"
 *
 * @param  max <description_of_max_param>
 * @param  c <description_of_c_param>
 * @param  distance <description_of_distance_param>
 * @return   <what_this_function returns>
 */
public static double allowableRange(double max, double c, double distance) {
...
}
```

This will allow VSCode to pick up on the documentation of this and allow anyone to understand what they need to pass to this function for it to function correctly.